### PR TITLE
Upgrade to LuckPerms 5 API

### PR DESCRIPTION
**Pull request by freddyheppell**
This now builds with no errors, however it is untested﻿

### CODE REVIEW

File: src/main/kotlin/com/projectcitybuild/spigot/modules/chat/listeners/ChatListener.kt [hunk 1]
```diff
@@ -29,30 +31,32 @@ class ChatListener : Listenable<AsyncPlayerChatEvent> {
...
+                .map(NodeType.INHERITANCE::cast)
+                .map { node -> node.value }
                 .collect(Collectors.toSet())
                 .joinToString(separator = "")
 
         val groups = mutableListOf<String>()
         groupNodes.forEach { group ->
             val group = group ?: return
-            val displayName = group.displayName ?: group.name
+            val displayName = group.groupName
```
Comment (freddyheppell) [inline src/main/kotlin/com/projectcitybuild/spigot/modules/chat/listeners/ChatListener.kt:56]: I wasn't able to figure out the distinction of display names and names in the new API, so for now this just uses `groupName`

Comment (freddyheppell) [reply src/main/kotlin/com/projectcitybuild/spigot/modules/chat/listeners/ChatListener.kt:56]: Looks like we might have to get the group and see if it has any `DisplayNameNode`s?

https://github.com/lucko/LuckPerms/wiki/Developer-API:-Usage#the-basics-of-node

**Comment by andyksaw** [comment]
Sorry for the long hiatus. I'm doing some local tests on this now

File: src/main/kotlin/com/projectcitybuild/spigot/modules/chat/listeners/ChatListener.kt (hunk 1 above)
Comment (andyksaw) [reply src/main/kotlin/com/projectcitybuild/spigot/modules/chat/listeners/ChatListener.kt]: It turns out you can just fetch the group from `groupManager` and directly access the `displayName` property 👍

**Comment by andyksaw** [comment]
All working. Thanks for the upgrades!

**Pull request merged**